src/components/entries/Tooltip.js

```diff
@@ -0,0 +1,138 @@
+import { useRef } from 'preact/hooks';
+import {
+  useEffect,
+  useState
+} from 'react';
+
+/**
+ * @param {Object} props
+ * @param {String} props.labelId
+ * @param {String} props.value
+ */
+export default function TooltipWrapper(props) {
+
+  if (!props.value) {
+    return props.children;
+  }
+
+  return <Tooltip { ...props } />;
+}
+
+function Tooltip(props) {
+  const {
+    value,
+    labelId
+  } = props;
+
+  const [ visible, setShow ] = useState(false);
+
+  let timeout = null;
+
+  const wrapperRef = useRef(null);
+  const tooltipRef = useRef(null);
+
+  const showTooltip = async (event) => {
+    const show = () => !visible && setShow(true);
+
+    if (event instanceof MouseEvent) {
+      timeout = setTimeout(show, 200);
+    } else {
+      show();
+    }
+  };
+
+  const hideTooltip = () => setShow(false);
+
+  const hideTooltipViaEscape = (e) => {
+    e.code === 'Escape' && hideTooltip();
+  };
+
+  const isTooltipHovered = ({ x, y }) => {
+    const tooltip = tooltipRef.current;
+    const wrapper = wrapperRef.current;
+
+    return tooltip && (
+      inBounds(x, y, wrapper.getBoundingClientRect()) ||
+      inBounds(x, y, tooltip.getBoundingClientRect())
+    );
+  };
+
+  useEffect(() => {
+    const { current } = wrapperRef;
+
+    if (!current) {
+      return;
+    }
+
+    const hideHoveredTooltip = (e) => {
+      if (visible && !isTooltipHovered({ x: e.x, y: e.y })) {
+        hideTooltip();
+      }
+    };
+
+    const hideFocusedTooltip = (e) => {
+      const { relatedTarget } = e;
+      const isTooltipChild = (el) => el && !!el.closest('.bio-properties-panel-tooltip');
+
+      if (visible && !isHovered(wrapperRef.current) && !isTooltipChild(relatedTarget)) {
+        hideTooltip();
+      }
+    };
+
+    document.addEventListener('focusout', hideFocusedTooltip);
+    document.addEventListener('mousemove', hideHoveredTooltip);
+
+    return () => {
+
+      document.removeEventListener('mousemove', hideHoveredTooltip);
+      document.removeEventListener('focusout', hideFocusedTooltip);
+    };
+  }, [ wrapperRef, visible ]);
```

Comment:
Minor: Refs are stable by design; we should rather care about `wrapperRef.current` in the dependencies.